integration-tests/contracts/ethereum_mercury_contracts.go

```diff
@@ -0,0 +1,224 @@
+	Address() string
+	CommitTrade(commitment [32]byte) error
+	ResolveTrade(encodedCommitment []byte) (string, error)
+	ResolveTradeWithReport(chainlinkBlob []byte, encodedCommitment []byte) (*types.Receipt, error)
+}
+
+type EthereumExchanger struct {
+	address   *common.Address
+	client    blockchain.EVMClient
+	exchanger *exchanger.Exchanger
+}
+
+func (v *EthereumExchanger) Address() string {
+	return v.address.Hex()
+}
+
+func (e *EthereumExchanger) CommitTrade(commitment [32]byte) error {
+	txOpts, err := e.client.TransactionOpts(e.client.GetDefaultWallet())
+	if err != nil {
+		return err
+	}
+	tx, err := e.exchanger.CommitTrade(txOpts, commitment)
+	if err != nil {
+		return err
+	}
+	return e.client.ProcessTransaction(tx)
+}
+
+func (e *EthereumExchanger) ResolveTrade(encodedCommitment []byte) (string, error) {
+	callOpts := &bind.CallOpts{
+		From:    common.HexToAddress(e.client.GetDefaultWallet().Address()),
+		Context: context.Background(),
+	}
+	data, err := e.exchanger.ResolveTrade(callOpts, encodedCommitment)
+	if err != nil {
+		return "", err
+	}
+	return data, nil
+}
+
+func (e *EthereumExchanger) ResolveTradeWithReport(chainlinkBlob []byte, encodedCommitment []byte) (*types.Receipt, error) {
+	txOpts, err := e.client.TransactionOpts(e.client.GetDefaultWallet())
+	txOpts.GasLimit = 1000000
+	if err != nil {
+		return nil, err
+	}
+	tx, err := e.exchanger.ResolveTradeWithReport(txOpts, chainlinkBlob, encodedCommitment)
+	if err != nil {
+		// blockchain.LogRevertReason(err, exchanger.ExchangerABI)
+		return nil, err
+	}
+	err = e.client.ProcessTransaction(tx)
+	if err != nil {
+		return nil, err
+	}
+	err = e.client.WaitForEvents()
+	if err != nil {
+		return nil, err
+	}
+	receipt, err := e.client.GetTxReceipt(tx.Hash())
+	if err != nil {
+		return nil, err
+	}
+	return receipt, nil
+}
+
+type VerifierProxy interface {
+	Address() string
+	InitializeVerifier(configDigest [32]byte, verifierAddress string) error
+	Verify(signedReport []byte) error
+}
+
+type EthereumVerifierProxy struct {
+	address       *common.Address
+	client        blockchain.EVMClient
+	verifierProxy *verifier_proxy.VerifierProxy
+}
+
+func (v *EthereumVerifierProxy) Address() string {
+	return v.address.Hex()
+}
+
+func (v *EthereumVerifierProxy) InitializeVerifier(configDigest [32]byte, verifierAddr string) error {
+	txOpts, err := v.client.TransactionOpts(v.client.GetDefaultWallet())
+	if err != nil {
+		return err
+	}
+	tx, err := v.verifierProxy.InitializeVerifier(txOpts, configDigest, common.HexToAddress(verifierAddr))
+	if err != nil {
+		return err
+	}
+	return v.client.ProcessTransaction(tx)
+}
+
+func (v *EthereumVerifierProxy) Verify(signedReport []byte) error {
+	txOpts, err := v.client.TransactionOpts(v.client.GetDefaultWallet())
+	if err != nil {
+		return err
+	}
+	tx, err := v.verifierProxy.Verify(txOpts, signedReport)
+	if err != nil {
+		return err
+	}
+	return v.client.ProcessTransaction(tx)
+}
+
+type Verifier interface {
```

Comment:
@smartcontractkit/test-tooling-team  @lukaszcl - wondering whether we should keep using these interface defining standard. Will these be used anywhere else? Why not directly writing the methods if it's just used by one struct?

Reply:
good question. I'd like to hear more about it from @smartcontractkit/test-tooling-team